insights/parsers/du_usage.py

```diff
@@ -0,0 +1,55 @@
+"""
+du - Command
+============
+
+Module for the processing of output from the ``du`` command.  The base class
+``DiskUsage`` provides all of the functionality for all classes.
+Data is avaliable as rows of the output contained
+in one ``Record`` object for each line of output.
+
+Sample input data for the ``du -sh`` command looks like::
+22G     /var/lib/mongodb
+
+"""
+
+from .. import Parser, parser, LegacyItemAccess
+
+
+def parse_du_lines(du_content):
+
+    du_output = {}
+    """Parse contents of each line in ``du`` output.
+    Parse each line of ``du`` output ensuring that wrapped lines are
+    reassembled prior to parsing, and that mount names containing spaces
+    are maintained.
+
+    Parameters:
+        du_content (list): Lines of du output to be parsed.
+
+    """
+    for line in du_content:
+        line_split = line.split()
+        du_output[line_split[1]] = line_split[0]
+
+    return du_output
+
+
+@parser("du_-h")
+class DiskUsage(LegacyItemAccess, Parser):
+    def __len__(self):
+        return len(self.data)
+
+    def parse_content(self, content):
+        self.data = parse_du_lines(content)
+
+    def get_size(self, dir_path):
+        return self.data[dir_path]
```

Comment:
You should check if the `dir_path` is in the `self.data` or not to avoid the `KeyError`.
Or use `return self.data.get(dir_path)` which will return None when `dir_path` does not exist.

Btw, seems the `get` of the `LegacyItemAccess` already provides the same functions to this `get_size`

Reply:
Done